src/dotnet/Core/Concurrency/InterlockedExt.cs

```diff
@@ -0,0 +1,16 @@
+namespace ActualChat.Concurrency;
+
+public static class InterlockedExt
+{
+    public static void ExchangeIfGreaterThan(ref long location, long value)
+    {
+        var current = Interlocked.Read(ref location);
+        while (current < value) {
```

Comment:
method name is `..IfGreaterThan` but in fact `IsLessThan`?

Reply:
Exchange if the value is greater than location - do you have suggestions to describe this logic?